app/controllers/signup_controller.rb

```diff
@@ -25,10 +21,12 @@ def start
     if request.post?
       handle_with(SignupStart,
                   existing_signup_state: signup_state,
+                  trusted_state: session[:trusted],
                   return_to: session[:return_to],
+                  session: self,
                   success: lambda do
                     save_signup_state(@handler_result.outputs.signup_state)
-                    redirect_to action: :verify_email
+                    redirect_to action: signup_state.after_email_action
```

Comment:
See other comment about making the choice of redirection based on more simple state in `SignupState` and not having `SignupState` know about action names.

Reply:
I think we still have to have signup_state (or something else) decide what the "after_email_action" should be.

The user may have changed the email in the signup form, in which case we can no longer trust it and it needs to be verified.

Or the user may change the role on signup, in which case we do need to allow them to fill out password

Reply:
i've moved the logic for this to the SignupStart handler